File: IT/Input-Output/Esercizi/Java/Esercizio27.java
/* file Esercizio27.java */

import java.util.Scanner;
import java.io.Console;
import java.io.File;
import java.io.FileWriter;
import java.io.BufferedWriter;
import java.io.IOException;

public class Esercizio27 {
	public static void main(String [] args) throws Exception {
		System.out.println("Un semplice editor di testo\nPremi \"Ctrl+C\" (su Windows) o \"Ctrl+D\" (Linux e MacOS) per finire di scrivere.\n");

		String filename = "";
		Scanner scanner = new Scanner(System.in);
		StringBuilder text = new StringBuilder();

		while (scanner.hasNextLine()) {
			String token = scanner.nextLine();
			text.append(token + System.lineSeparator());
		}

		boolean quit = false;
		Console console = System.console();

		while (!quit) {
			String buff = console.readLine("Digita \"s\" per salvare su file, \"q\" per uscire senza salvare: ");

			if (buff.length() > 1) continue;
			char opt = Character.toLowerCase(buff.charAt(0));

			if (opt == 's') {
				filename = console.readLine("Salva con nome: ");

				try {
					File doc = new File(filename);
					FileWriter writer = new FileWriter(doc);
					BufferedWriter buffer = new BufferedWriter(writer);

					String [] lines = text.toString().split(System.lineSeparator());

					for (String line : lines) {
						buffer.write(line + System.lineSeparator());
					}

					buffer.close();

					System.out.println("File \"" + doc.getName() + "\" salvato su disco.");
					quit = true;
				}
				catch (IOException e) {
					e.printStackTrace();
				}
				/*
				finally {
					scanner.close();
				}
				*/
			}
			else if (opt == 'q') {
				quit = true;
			}
		}
	}
}

File: IT/Input-Output/Esercizi/C++/esercizio27.cpp
/*
file esercizio27.cpp
compilare con: g++ -std=c++11 esercizio27.cpp
*/

#include <iostream>
#include <fstream>

using std::cin;
using std::cout;
using std::endl;
using std::string;
using std::ofstream;

int main()
{
	cout << "Un semplice editor di testo\nPremi \"Ctrl+C\" (su Windows) o \"Ctrl+D\" (Linux e MacOS) per finire di scrivere." << endl << endl;

	string text;
	string line;

	while (!getline(cin, line).eof()) {
		text += line + "\n";
	}

	cin.clear();

	bool quit = false;

	while (!quit) {
		cout << "Digita \"s\" per salvare su file, \"q\" per uscire senza salvare: ";

		string buff;
		getline(cin, buff);

		if (buff.size() > 1) continue;

		char opt = tolower(buff[0]);

		if (opt == 's') {
			cout << "Salva con nome: ";
			string filename;
			getline(cin, filename);

			ofstream doc(filename);
			doc << text;

			cout << "File \"" << filename << "\" salvato su disco." << endl;
			quit = true;
		}
		else if (opt == 'q') {
			quit = true;
		}
	}
}

